service/controller/key/cluster.go

```diff
@@ -115,6 +115,17 @@ func MasterInstanceType(cluster v1alpha1.Cluster) string {
 	return clusterProviderSpec(cluster).Provider.Master.InstanceType
 }
 
+func ManagedRecordSets(cluster v1alpha1.Cluster) []string {
+	tcBaseDomain := TenantClusterBaseDomain(cluster)
+	return []string{
+		fmt.Sprintf("%s.", tcBaseDomain),
+		fmt.Sprintf("\\052.%s.", tcBaseDomain), // \\052 - `*` wildcard record
+		fmt.Sprintf("api.%s.", tcBaseDomain),
+		fmt.Sprintf("etcd.%s.", tcBaseDomain),
+		fmt.Sprintf("internal-api.%s.", tcBaseDomain),
```

Comment:
Also can't we list services and check annotations to discover entries that need to be removed? That way we won't need to extend this list in case we add something else.

Reply:
I've looked into that option. But we anyway need to know hosted zone to remove the record. Also there might be other apps adding records based on different annotations. So from that point of view we would need to extend list with annotations in the same way as  list of managed recordsets

Reply:
But you can check if the cluster base domain matches with the value of annotation. Wouldn't that be sufficient?

Reply:
But you need to know about those annotations. Fo external-dns it will be something like `external-dns-foo`,  so you'll look into this in operator. Then we add later another app which makes it 'other-app-foo-annotation' - what to do - update operator again? Cleaning route53 directly makes it app independent. Relying on annotations brings scope we don't know how will look later

Reply:
Okay. I don't know enough about external-dns but I bet it's discoverable. Otherwise how could external-dns work. One way or another it's not independent. I won't block on that though. Let's sort the wildcard and go.

Reply:
It is out of external-dns (or any other app) scope. It is just about bringing recordsets to the state, managed by CF